drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -68,6 +68,10 @@ struct aci_set_ble_addr {
 	uint8_t value[6];
 } __packed;
 
+#if defined(CONFIG_BT_HCI_SETUP)
```

Comment:
Since you're unconditionally selecting this option (currently indirectly by selecting `BT_HCI_SET_PUBLIC_ADDR`), there's no way that the driver could get built without it, i.e. you should be able to remove all the `#if defined(CONFIG_BT_HCI_SETUP)` stuff.

Reply:
Indeed, good remark